### PR TITLE
add dependency and RHEL to meta.yml

### PR DESCRIPTION
**Pull request by martbhell**
Hi!

This looks useful, thanks for the effort!
I've been using this on CentOS7 and it works so added RedHat version.
Added the dependencies to meta file so one can install this role with ansible-galaxy
parametrized parts of the config - to change subject line
added rsa:4096 and -sha256 to the self generated request

// Johan

### CODE REVIEW

**Comment by colstrom** [comment]
Looks good to me. Thanks for contributing!

File: tasks/main.yml [hunk 1]
```diff
@@ -18,6 +18,17 @@
   tags:
     - ssl-certificate
 
+- name: Set chmod on private key
+  file: 
+    path: "{{ ssl_certificate_path_private }}" 
+    mode: "0440"
```
Comment (stevenharradine) [inline tasks/main.yml]: Is this too open can it be `0400`?

Comment (martbhell) [inline tasks/main.yml]: 0400 is probably fine

Comment (stevenharradine) [inline tasks/main.yml]: Everything else was 0440 but im testing 0400 in my personal fork and it seems to work for nginx and haproxy not sure what your using but I will likely push that by the end of the week

**Pull request merged**